python/python/lance/dataset.py

```diff
@@ -1047,6 +1048,31 @@ def _coerce_reader(
         )
 
 
+def _validate_schema(schema: pa.Schema):
+    """
+    Make sure the metadata is valid utf8
```

Comment:
Add a comment for 'expecting raising a ValueError if the schema metadata is not valid'?